formats/cbor/commonMain/src/kotlinx/serialization/cbor/internal/Encoding.kt

```diff
@@ -428,6 +451,88 @@ internal class CborDecoder(private val input: ByteArrayInput) {
         return result
     }
 
+    /**
+     * Skips the current value element. Bytes are processed to determine the element type (and corresponding length), to
+     * determine how many bytes to skip.
+     *
+     * For primitive (finite length) elements (e.g. unsigned integer, text string), their length is read and
+     * corresponding number of bytes are skipped.
+     *
+     * For elements that contain children (e.g. array, map), the child count is read and added to a "length stack"
+     * (which represents the "number of elements" at each depth of the CBOR data structure). When a child element has
+     * been skipped, the "length stack" is [pruned][prune]. For indefinite length elements, a special marker is added to
+     * the "length stack" which is only popped from the "length stack" when a CBOR [break][isEnd] is encountered.
+     */
+    fun skipElement() {
+        if (curByte == -1) throw SerializationException("Missing element value")
+        val lengthStack = mutableListOf<Int>()
+
+        do {
+            if (isIndefinite()) {
+                lengthStack.add(LENGTH_STACK_INDEFINITE)
+            } else if (isEnd()) {
+                if (lengthStack.removeLastOrNull() != LENGTH_STACK_INDEFINITE)
+                    throw CborDecodingException("next data item", curByte)
+                prune(lengthStack)
+            } else {
+                val header = curByte and 0b111_00000
+                val length = elementLength()
+                if (header == HEADER_ARRAY || header == HEADER_MAP) {
+                    if (length > 0) lengthStack.add(length)
+                } else {
+                    input.skip(length)
+                    prune(lengthStack)
+                }
+            }
+
+            if (readByte() == -1 && lengthStack.isNotEmpty()) throw SerializationException("EOF while skipping element")
```

Comment:
```suggestion
            if (readByte() == -1 && lengthStack.isNotEmpty()) throw CborDecodingException("Unexpected EOF while skipping element")
```

This error usually caused by a malformed input, not a general problems with serialization framework

Reply:
🤦  I totally missed the `CborDecodingException` constructor that takes a `message` (and doesn't require `curByte`).